File: server/php/src/Application/Domain/Payments/PaymentDetails/CustomerDeviceOutput.php
<?php

namespace OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails;

final class CustomerDeviceOutput
{
    public ?string $ipAddressCountryCode = null;

    public function toArray(): array
    {
        return [
            'ipAddressCountryCode' => $this->ipAddressCountryCode,
        ];
    }
}

File: server/ruby/app/business/domain/payments/payment_details/customer_device_output.rb
module Business
  module Domain
    module Payments
      module PaymentDetails
        class CustomerDeviceOutput
          attr_accessor :ip_address_country_code

          def initialize(ip_address_country_code: nil)
            @ip_address_country_code = ip_address_country_code
          end

          def to_h
            {
              'ipAddressCountryCode' => @ip_address_country_code
            }
          end
        end
      end
    end
  end
end
